docs/equality.md

```diff
@@ -0,0 +1,348 @@
+* Perf: User expects full performance down to native
+* Compilation today: compiles to IL instruction ✅
+* Perf today: good ✅
+* [sharplab float32](https://sharplab.io/#v2:DYLgZgzgNAJiDUAfYBTALgAjBgFADxC2AHsBDNAZgCYBKXAT0LBPOroF5c8NP6aBYAFBA===)
+
+### primitive `string`, `decimal`
+
+* Semantics: .NET equivalent equality, non-localized for strings
+* Perf: User expects full performance down to native
+* Compilation today: compiles to `String.Equals` or `Decimal.op_Equality` call ✅
+* Perf today: good ✅
+* [sharplab decimal](https://sharplab.io/#v2:DYLgZgzgNALiCWwoBMQGoA+wCmMAEYeAFAB4h7LYDG8AtgIbACUxAnuZTQ83gLzEk+eVkwCwAKCA)
+* [sharplab string](https://sharplab.io/#v2:DYLgZgzgNALiCWwoBMQGoA+wCmMAEYeAFAB4h4QwBO8AdgOYCUxAnuZTQ8wLzEl68WjALAAoIA==)
+
+### tuple type (size <= 5)
+
+* Semantics: User expects structural
+* Perf: User expects flattening to constituent checks
+* Compilation today: tuple equality is flattened to constituent checks up to size 5 ✅
+* Perf today: ok up to size 5 ✅
+* [sharplab (int * double * 'T), with example reductions/optimizations noted](https://sharplab.io/#v2:DYLgZgzgPgsAUMApgFwARlQCgB4iwSwDs0AqVAEwHsBXAIyVTIHIAVASjdQE9UBeLbH25t48TCVFxB/LpIC0cosCJEA5goB8kgOKJCiAE74AxgFEAjtQCGy5D0Gy48BUpWF1crU7gAJKxAALAGFKAFsABysDRAA6XX0jM0sbfDsAMX80B1R5RUJlQjVNHT1DEwtrWy4ASWIjQggTAB4WAEZGVBYAJg6WAGYNVAdcgHlw5HxQ/AAvQ00sckQAN3wDNHiypMrUmrqiRuMRbwyIZAqbCBZqcKQ+1AAZK3drVUQABSMpiaXECDjSxIhCJRQwCVoAGmwXUhfU4mC4EK40K4sNyrkK7mK3iQaGMYUi0QMQkezysrw+k1S+B+fw2gPxIIM8Dp5WSVQA6qlggzCSdcTzQdh2gjUAAyUXMgGs7Z2TnIbnA3mZVB4xWCnpIsUSuAsrYpWVcoEEwx8lUConYO4o3KDSQ4s1qon8EmqF7vT5Umn/BImI2M+DGRDmIbC9rigNBoYanrhnVSvUcw3m2rIeoHB3Gi1WvqSEhHeBAA==)
+
+### tuple type (size > 5)
+
+* Semantics: User expects structural
+* Perf: User expects flattening to constituent checks
+* Compilation today: compiled to GenericEqualityIntrinsic
+* Perf today: size > 5 is not flattened. This means the check does type tests, does virtual calls via IStructuralEqualityComparer, boxes etc. ❌(Problem3)
+* [sharplab for size 6](https://sharplab.io/#v2:DYLgZgzgPgsAUMApgFwARlQCgB4iwSwDs0AqVI0841MimqyigSidQE9UBeLbL9p+EA==)
+
+### struct tuple type
+
+* Semantics: User expects structural
+* Perf: User expects flattening to constituent checks, **or at least the same optimizations as tuples**
+* Compilation today: compiled to GenericEqualityIntrinsic
+* Perf today: boxes, does type tests, does virtual calls via IStructuralEqualityComparer etc. ❌(Problem4)
+* [sharplab for size 3](https://sharplab.io/#v2:DYLgZgzgPgsAUMApgFwARlQCgB4lRZAJwFcBjNTASwDs0AqVG+x2gSldQE9UBeLbXl1bwgA=)
+
+### array type (byte[], int[], some-struct-type[],  ...)
+
+* Semantics: User expects structural
+* Perf: User expects perf is sum of constituent parts
+* Compilation today: GenericEqualityIntrinsic<uint8[]>
+* Perf today:  this is hand-optimized for some primitive element types ✅ but boxes each element if "other" is struct or T, see Problem1, Problem2 ❌ 
```

Comment:
`some primitive` is mentioned often enough to justify for a quick URL into the place of the code where a reader can see the list of manually optimized primitives.

Reply:
Sure, added a link